qingstor/qsctl/utils.py

```diff
@@ -30,6 +30,70 @@
 )
 
 
+class UploadIdRecorder(object):
+    """
+    This class stores upload_id for uploading a large object via multipart API.
+
+    - In-memory record dict:
+    key: <local_path>|<bucket>/<key> value: <upload_id>
+
+    - On-disk record format:
+    <local_path>|<bucket>/<key>#<upload_id>\n
+    """
+
+    def __init__(self, record_filename):
+        self.separator = "|"
+        self.records = {}
+        self.dirty = False
+
+        if os.path.exists(record_filename):
+            self.file = open(record_filename, "r+")
+            # Load records from file.
+            records = self.file.readlines()
+            for record in records:
+                if is_python2:
+                    record = record.decode("utf-8")
+                kv = record.rsplit("#", 1)
+                if len(kv) == 2:
+                    # Remove the trailing \n
+                    key = kv[1][:-1]
+                    self.records[kv[0]] = key
+        else:
+            self.file = open(record_filename, "w+")
+
+    def put_record(self, local_path, bucket, key, upload_id):
+        key = self._get_record_key(local_path, bucket, key)
```

Comment:
Do `getcwd` in Recoder may be a better idea, change local_path directly may affect other functions.